monkeytype/typing.py

```diff
@@ -501,6 +502,64 @@ def rewrite_Generator(self, typ):
         return typ
 
 
+class RewriteMostCommonBase(TypeRewriter):
+    """
+    Relaces a union of classes by the least common
+    base of its members, provided that
+    parent classes of members are unique leading
+    up to the `object` class (no double inheritance)
+    """
+
+    def compute_bases(self, klass):
+        """
+        Compute list of bases of a given class,
+        starting from (but exluding) the root `object`,
+        ending  (and including) the actual class.
+        """
+        bases = []
+
+        curr_klass = klass
+
+        while curr_klass is not object:
+            bases.append(curr_klass)
+
+            curr_bases = curr_klass.__bases__
+
+            if len(curr_bases) != 1:
+                break
+
+            curr_base = curr_bases[0]
+            curr_klass = curr_base
+        return bases[::-1]
+
+    def merge_common_bases(self, first_bases, second_bases):
+        merged_bases = []
+
+        min_len = min(len(first_bases), len(second_bases))
+
+        first_bases = first_bases[:min_len]
+        second_bases = second_bases[:min_len]
```

Comment:
These lines seem unnecessary, as they just duplicate what is already the default behavior of `zip` (if the given iterables are not the same length, "extra" elements in the longer one are ignored)